autoload/go/fmt.vim

```diff
@@ -52,13 +52,25 @@ endif
 "  this and have VimL experience, please look at the function for
 "  improvements, patches are welcome :)
 function! go#fmt#Format(withGoimport)
-    " save cursor position, folds and many other things
-    let l:curw = {}
-    try
-        mkview!
-    catch
+    if g:go_fmt_experimental == 1
+        " Using winsaveview to save/restore cursor state has the problem of
+        " closing folds on save:
+        "   https://github.com/fatih/vim-go/issues/502
+        " One fix is to use mkview instead. Unfortunately, this sometimes causes
+        " other bad side effects:
+        "   https://github.com/fatih/vim-go/issues/728
+        " and still closes all folds if foldlevel>0:
+        "   https://github.com/fatih/vim-go/issues/732
+        let l:curw = {}
+        try
+            mkview!
+        catch
+            let l:curw=winsaveview()
+        endtry
+    else
```

Comment:
I think what we want to do is using `winsaveview()` in normal mode and `mkview`. Otherwise we lose cursor position and also doesn't align with the old version.